README.md

```diff
@@ -47,13 +47,51 @@ You can also set `-Zmiri-start-fn` to make Miri start evaluation with the
 
 ## Running Miri on your own project('s test suite)
 
-Install Miri as a cargo subcommand with `cargo install --all-features`, and install
-a full libstd as described above.
+Install Miri as a cargo subcommand with `cargo install --all-features --path .`.
 
-Then, inside your own project, use `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly
-miri` to run your project, if it is a bin project, or run
-`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri test` to run all tests in your
-project through Miri.
+Compile your project and its dependencies against a MIR-enabled libstd as described
+above:
+
+1. Run `cargo clean` to eliminate any cached dependencies that were built against
+the non-MIR `libstd`.
+2. To run all tests in your project through, Miri, use
+`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri test`.
+3. If you have a binary project, you can run it through Miri using
+`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri`.
+
+### Common Problems
+
+When modifying the above instructions, you may encounter a number of confusing compiler
+errors.
+
+#### "constant evaluation error: no mir for `<function>`"
+
+You may have forgotten to set `MIRI_SYSROOT` when calling `cargo miri test`, and
+your program called into `std` or `core`. Be sure to set `MIRI_SYSROOT=~/.xargo/HOST`.
+
+#### "found possibly newer version of crate `std` which `<dependency>` depends on"
+
+Your build directory may contain artifacts from an earlier build that did/did not
+have `MIRI_SYSROOT` set. Run `cargo clean` before switching from non-Miri to Miri
+builds and vice-versa.
+
+#### "found crate `std` compiled by an incompatible version of rustc"
```

Comment:
I do not understand how/why this is different from the one right before. Seems to me both are caused by "sysroot confusion", and both are solved the same way. Why should we give different advise for these two?

Reply:
I don't understand. They have different causes and different solutions.

Reply:
Oh, the difference is "deps were built without `MIRI_SYSROOT` vs `MIRI_SYSROOT` was built with the wrong toolchain, I see.  Makes sense.

I think what threw me off was specifically mentioning `cargo miri test`. The same applies to `cargo miri`, right? So maybe just say `cargo miri` then.